url75694980d5_fabioalmeid_tp01ftc_tgz-pJ8-agentes_jade_Centralizador_ExemploParserGramaticaCentralizadorJ8/src/util/GeradorAleatorioMsg.java

```diff
@@ -0,0 +1,191 @@
+package util;
+
+
+
+import gramatica.Centralizador.Absyn.Dados;
+import gramatica.Centralizador.Absyn.EApply1;
+import gramatica.Centralizador.Absyn.EApply2;
+import gramatica.Centralizador.Absyn.ECollect1;
+import gramatica.Centralizador.Absyn.ECollect2;
+import gramatica.Centralizador.Absyn.EData1;
+import gramatica.Centralizador.Absyn.EData2;
+import gramatica.Centralizador.Absyn.EData3;
+import gramatica.Centralizador.Absyn.EData4;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import agentes.jade.Centralizador.TarefaCentralizador;
+
+public class GeradorAleatorioMsg {
+	private static final int DATA_QUANTITY = 5;
+	
+	public static String getRandomStartMeasure() {
+		TarefaCentralizador tc = new TarefaCentralizador();
+		
+		// seta iniciar medicao
+		tc.setAcao(new ECollect1());
+
+		int iRandom = (int) (Math.random() * (DATA_QUANTITY));
+		switch (iRandom) {
+		case 0:
+			tc.setDados(new EData1());
+			break;
+		case 1:
+			tc.setDados(new EData2());
+			break;
+		case 2:
+			tc.setDados(new EData3());
+			break;
+		case 3:
+			tc.setDados(new EData4());
+			break;
+		case 4:
+			int iRandom2 = (int) (Math.random() * (DATA_QUANTITY-1));
+			int iRandom3 = (int) (Math.random() * (DATA_QUANTITY-1));
+			
+			Map <Integer, Dados> hm = new HashMap<Integer, Dados>();
+			hm.put(0, new EData1());
+			hm.put(1, new EData2());
+			hm.put(2, new EData3());
+			hm.put(3, new EData4());
+			
+			tc.setDados(hm.get(iRandom2));
+			if (iRandom3 != iRandom2)
+				tc.setDados(hm.get(iRandom3));
+							
+			break;
+		default:
+			System.out.println("Error: Index out of range on TarefaCentralizador");
+			break;
+		}		
+		return tc.prettyPrinterTarefa();
+	}
+	
+	public static String getRandomStartMeasureWithErrors() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 46 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 98% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.